L1Trigger/TrackerDTC/src/Setup.cc

```diff
@@ -476,13 +478,176 @@ namespace trackerDTC {
     }
   }
 
+  // configure TPSelector
+  void Setup::configureTPSelector() {
+    // configure TrackingParticleSelector
+    const double ptMin = tpMinPt_;
+    constexpr double ptMax = 9.e9;
+    const double etaMax = tpMaxEta_;
+    const double tip = tpMaxVertR_;
+    const double lip = tpMaxVertZ_;
+    constexpr int minHit = 0;
+    constexpr bool signalOnly = true;
+    constexpr bool intimeOnly = true;
+    constexpr bool chargedOnly = true;
+    constexpr bool stableOnly = false;
+    tpSelector_ = TrackingParticleSelector(
+        ptMin, ptMax, -etaMax, etaMax, tip, lip, minHit, signalOnly, intimeOnly, chargedOnly, stableOnly);
+    tpSelectorLoose_ = TrackingParticleSelector(ptMin, ptMax, -etaMax, etaMax, tip, lip, minHit, false, false, false, stableOnly);
+  }
+
+  //
+  int Setup::layerId(const TTStubRef& ttStubRef) const {
+    const DetId& detId = ttStubRef->getDetId();
+    return detId.subdetId() == StripSubdetector::TOB ? trackerTopology_->layer(detId)
+                                                     : trackerTopology_->tidWheel(detId) + offsetLayerDisks_;
+  }
+
+  //
+  int Setup::barrel(const TTStubRef& ttStubRef) const {
+    const DetId& detId = ttStubRef->getDetId();
+    return detId.subdetId() == StripSubdetector::TOB;
+  }
+
+  //
+  int Setup::psModule(const TTStubRef& ttStubRef) const {
+    const DetId& detId = ttStubRef->getDetId();
+    return trackerGeometry_->getDetectorType(detId) == TrackerGeometry::ModuleType::Ph2PSP;
+  }
+
+  //
+  TTBV Setup::layerMap(const vector<int>& ints) const {
+    TTBV ttBV;
+    for (int layer = numLayers_ - 1; layer >= 0; layer--) {
+      const int i = ints[layer];
+      ttBV += TTBV(i, kfWidthLayerCount_);
+    }
+    return ttBV;
+  }
+
+  //
+  TTBV Setup::layerMap(const TTBV& hitPattern, const vector<int>& ints) const {
+    TTBV ttBV;
+    for (int layer = numLayers_ - 1; layer >= 0; layer--) {
+      const int i = ints[layer];
+      ttBV += TTBV((hitPattern[layer] ? i - 1 : 0), kfWidthLayerCount_);
+    }
+    return ttBV;
+  }
+
+  //
+  vector<int> Setup::layerMap(const TTBV& hitPattern, const TTBV& ttBV) const {
+    TTBV bv(ttBV);
+    vector<int> ints(numLayers_, 0);
+    for (int layer = 0; layer < numLayers_; layer++) {
+      const int i = bv.extract(kfWidthLayerCount_);
+      ints[layer] = i + (hitPattern[layer] ? 1 : 0);
+    }
+    return ints;
+  }
+
+  //
+  vector<int> Setup::layerMap(const TTBV& ttBV) const {
+    TTBV bv(ttBV);
+    vector<int> ints(numLayers_, 0);
+    for (int layer = 0; layer < numLayers_; layer++)
+      ints[layer] = bv.extract(kfWidthLayerCount_);
+    return ints;
+  }
+
+  //
+  double Setup::dZ(const TTStubRef& ttStubRef, double cot) const {
+    const DetId& detId = ttStubRef->getDetId();
+    SensorModule* sm = sensorModule(detId + 1);
+    const double sigma = sm->pitchCol() * sm->tiltCorrection(cot);
+    const double digi = tmttBaseZ_;
+    const double dZ = sigma + digi;
```

Comment:
Add comment "Uncertainty components added linearly (simpler)"

Reply:
that has nothing to do with simpler

Reply:
The mathemetically correct way to add uncertainties is in quadrature. So why are they being added linearly if it is not to save CPU/resources?

Reply:
I disagree, maybe I should not call it uncertainty though.